lib/util/apis.dart

```diff
@@ -281,3 +284,24 @@ Future<Map<String, dynamic>> followAndUnfollow(String otherUser, String token) a
     },
   );
 }
+
+Future<Map<String, dynamic>> checkIn(String communityName, String token) async {
```

Comment:
Why is this happening here? Why isn't it happening within the `createPost` function itself? 

I don't think there's any reason to go like this:
1. Make request from client to server to create post
2. Create post on server
3. Return response from server
4. Make request from client to server to check in
5. Create check-in record on server (or not, if user has already checked in the community today)
6. Return response from server

Since this is a workflow that has to happen on every post creation (checking wether or not we should create a check-in record and creating that record if we need to), we can skip steps 3 and 4.

Reply:
I do somewhat agree with you but this in essence is a 2 step process and should be like that. In the off chance that a post isn't successful then there is no reason for the check in to even occur.